tmt/steps/report/junit.py

```diff
@@ -58,40 +69,188 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
+        """ Returns results of all executed tests """
+        return [r for r in self._results if r.result != ResultOutcome.INFO]
+
+    @functools.cached_property
+    def skipped(self) -> list[tmt.Result]:
+        """ Returns results of skipped tests """
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self) -> list[tmt.Result]:
+        """ Returns results of failed tests """
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self) -> list[tmt.Result]:
+        """ Returns results of tests with error/warn outcome """
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self) -> int:
+        """ Returns the total duration of all tests in seconds """
+        return sum([duration_to_seconds(r.duration) or 0 for r in self._results])
+
+
 def make_junit_xml(
-        report: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]'
-        ) -> 'junit_xml.TestSuite':
-    """ Create junit xml object """
-    junit_xml = import_junit_xml()
+        phase: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]',
+        flavor: str = DEFAULT_FLAVOR_NAME,
+        template_path: Optional[Path] = None,
+        include_output_log: bool = True,
+        prettify: bool = True,
+        results_context: Optional[ResultsContext] = None,
+        **extra_variables: Any
+        ) -> str:
+    """
+    Create JUnit XML file and return the data
+
+    :param phase: instance of a ReportPlugin.
+    :param flavor: name of a JUnit flavor to generate.
+    :param template_path: if set, the provided template will be used instead of
+        a pre-defined flavor template. In this case, the ``flavor`` must be set
+        to ``custom`` value.
+    :param include_output_log: if enabled, the ``<system-out>`` tags are included
+        in the final template output.
+    :param prettify: allows to control the XML pretty print.
+    :param results_context: if set, the provided ResultsContext is used in a template.
+    :param extra_variables: if set, these variables get propagated into the
+        Jinja template.
+    """
+
+    # Get the template context for TMT results
+    results_context = results_context or ResultsContext(phase.step.plan.execute.results())
 
-    suite = junit_xml.TestSuite(report.step.plan.name)
+    # Prepare the template environment
+    environment = default_template_environment()
 
-    for result in report.step.plan.execute.results():
+    template_dir = Path('steps/report/junit/templates/')
+    template_path = template_path or tmt.utils.resource_files(
+        template_dir / Path(f'{flavor}.xml.j2'))
+
+    # Use a FileSystemLoader for a non-custom flavor
+    if flavor != CUSTOM_FLAVOR_NAME:
```

Comment:
No, I am checking the `custom` flavor is not used. The **non-custom** flavors will use a Jinja2 `FileSystemLoader` to load a whole tmt folder with non-custom flavor jinja2 templates.

I want to be able to use jinja's template "includes" and template inheritance (blocks) for non-custom flavors and I want to disallow this behavior for `custom` flavors. I want to force the users to use self-contained custom templates (by this I mean the whole template defined in one file).

If we wanted to allow users to use template inheritance (`FileSystemLoader`), it would require getting a template directory (as a CLI parameter) and also pre-define some custom template entry point (which must always exist in this folder) which will be rendered by tmt (e.g. `custom.j2.xml`).

Reply:
from the definition it only allows `default` and `custom` flavours, that's the issue here I was pointing out (and not specifying fully)

so maybe changing the definition is needed?